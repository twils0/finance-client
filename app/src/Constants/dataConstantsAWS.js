// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
