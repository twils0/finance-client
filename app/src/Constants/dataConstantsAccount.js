export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
