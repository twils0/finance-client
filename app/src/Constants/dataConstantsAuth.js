export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
