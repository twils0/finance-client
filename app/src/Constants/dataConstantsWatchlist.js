export const actionTypes = {
  SET_WATCHLIST_STATUS: 'DATA_WATCHLIST_SECURITIES_STATUS',
  SET_SECURITIES_CURRENT: 'DATA_WATCHLIST_SET_SECURITIES_CURRENT',
  SET_SECURITIES_LIST: 'DATA_WATCHLIST_SET_SECURITIES_LIST',
  SET_SECURITIES_ALL: 'DATA_WATCHLIST_SET_SECURITIES_ALL',
  SET_SECURITY: 'DATA_WATCHLIST_SET_SECURITY',
  SET_SECURITY_DATA: 'DATA_WATCHLIST_SET_SECURITY_DATA',
  SET_SECURITY_SUBSCRIBE: 'DATA_WATCHLIST_SET_SECURITY_SUBSCRIBE',
  RESET_WATCHLIST_STATE: 'DATA_ACCOUNT_RESET_WATCHLIST_STATE',
};

export const codeTypeNames = {
  CONFIRM_PHONE: 'confirmPhone',
  CONFIRM_EMAIL: 'confirmEmail',
};

export const statusNames = {
  SECURITIES: 'watchlistStatusSecurities',
  GET_SECURITY_DATA: 'watchlistStatusGetSecurityData',
  UPDATE_SECURITIES: 'watchlistStatusUpdateSecurities',
};
