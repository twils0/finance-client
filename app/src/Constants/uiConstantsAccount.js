export const actionTypes = {
  SET_CURRENT_FORM: 'ACCOUNT_SET_CURRENT_FORM',
  SET_FORM_HEIGHT: 'ACCOUNT_SET_FORM_HEIGHT',
  SET_FORM_EDIT: 'ACCOUNT_SET_FORM_EDIT',
  SET_INPUT_VALUE_ERROR: 'ACCOUNT_SET_INPUT_VALUE_ERROR',
  SET_BUTTON_TEXT: 'ACCOUNT_SET_BUTTON_TEXT',
  SET_BUTTON_VISIBLE: 'ACCOUNT_SET_BUTTON_VISIBLE',
};

export const formNames = {
  CODE: 'codeForm',
  PROFILE: 'profileForm',
  BILLING: 'billingForm',
  CHANGE_PASSWORD: 'changePasswordForm',
  DELETE_ACCOUNT: 'deleteAccountForm',
};

export const buttonNames = {
  PROFILE: 'profileButton',
  BILLING: 'billingButton',
  CHANGE_PASSWORD: 'changePasswordButton',
  DELETE_ACCOUNT: 'deleteAccountButton',
  FIRST: 'firstButton',
  SECOND: 'secondButton',
};

export const buttonTexts = {
  RESEND: 'Resend',
  SUBMIT: 'Submit',
  EDIT: 'Edit',
  UPDATE: 'Update',
  CANCEL: 'Cancel',
  DELETE: 'Delete',
  CONFIRM: 'Confirm',
};

export const inputNames = {
  [formNames.CODE]: {
    CODE_PHONE: 'codeCodePhone',
  },
  [formNames.PROFILE]: {
    NAME: 'profileName',
    EMAIL: 'profileEmail',
    EMAIL_ADDITIONAL: 'profileEmailAdditional',
    PHONE: 'profilePhone',
  },
  [formNames.BILLING]: {
    NAME_ON_CARD: 'billingNameOnCard',
    PROMO_CODE: 'billingPromoCode',
    STRIPE: 'billingStripe',
  },
  [formNames.CHANGE_PASSWORD]: {
    OLD_PASSWORD: 'changePasswordOldPassword',
    NEW_PASSWORD: 'changePasswordNewPassword',
    NEW_PASSWORD2: 'changePasswordNewPassword2',
  },
  [formNames.DELETE_ACCOUNT]: {
    CONFIRM: 'deleteAccountConfirm',
  },
};
