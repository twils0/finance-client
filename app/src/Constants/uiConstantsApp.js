export const actionTypes = {
  SET_FONTS_STATUS: 'APP_SET_FONTS_STATUS',
  LOAD_FONTS: 'APP_LOAD_FONTS',
  SET_IMAGE_STATUS: 'APP_SET_IMAGE_STATUS',
  LOAD_IMAGE: 'APP_LOAD_IMAGE',
  SET_STRIPE_STATUS: 'APP_SET_STRIPE_STATUS',
  SET_STRIPE_ELEMENT_LOADED: 'APP_SET_STRIPE_ELEMENT_LOADED',
  LOAD_STRIPE: 'APP_FETCH_STRIPE',
  SET_STRIPE_OBJECT: 'APP_SET_STRIPE_OBJECT',
};

export const motionStatusTypes = {
  IDLE: 'IDLE',
  ACTIVE: 'ACTIVE',
};

export const imageNames = {
  EXAMPLE_HEADER: 'exampleHeader',
};

export const errorMessages = {
  NO_EMAIL: 'Please enter your email address.',
  INVALID_EMAIL: 'Please enter a valid email address.',
  NO_PASSWORD: 'Please enter your password.',
  NO_NEW_PASSWORD: 'Please enter your new password.',
  INVALID_PASSWORD:
    'Please enter a password containing at least 8 characters, one of the following special characters, $@!%*#?&, one letter, and one number.',
  NO_MATCH_PASSWORD: 'This password does not match your password above.',
  NO_NAME_ON_CARD: 'Please enter the name shown on your credit card.',
  NO_STRIPE: 'Please enter your credit card information.',
  INVALID_CARD_NUMBER: 'Please enter a valid credit card number.',
  INCOMPLETE_CARD_NUMBER: 'Please enter your full credit card number.',
  INVALID_CARD_EXPIRATION_DATE: 'Please enter a valid expiration date.',
  INVALID_CARD_CVC: 'Please enter a valid CVC.',
  INVALID_CARD_ZIP_CODE: 'Please enter a valid zip code.',
  NO_NAME: 'Please enter your name.',
  NO_PHONE: 'Please enter your mobile phone number.',
  INVALID_PHONE: 'Please enter a valid mobile phone number.',
  NO_VERIFICATION_CODE_PHONE: 'Please enter the verification code sent to your mobile phone.',
  NO_VERIFICATION_CODE_EMAIL: 'Please enter the verification code sent to your email.',
  NO_CONFIRMATION_CODE_PHONE: 'Please enter the confirmation code sent to your mobile phone.',
  INVALID_VERIFICATION_CODE:
    'Please verify you have entered the correct verification code or request a new code.',
  INVALID_CONFIRMATION_CODE:
    'Please verify you have entered the correct confirmation code or request a new code.',
  EXPIRED_VERIFICATION_CODE: 'This verification code has expired.',
  EXPIRED_CONFIRMATION_CODE: 'This confirmation code has expired.',
  EMAIL_IN_USE:
    'An account using this email already exists. Please login or use a different email.',
  EMAIL_IN_USE_INTERNAL:
    'An account using this email already exists. Please use a different email.',
  EMAIL_NOT_FOUND:
    'This email is not associated with an account. Please enter another email or sign up.',
  PASSWORD_NOT_FOUND: 'This password does not match our records.',
  PROMO_CODE_INVALID: 'Please enter a valid promo code.',
  PROMO_CODE_EXPIRED: 'This promo code is no longer valid.',
  RETRY_SIGN_UP: 'Please click "Cancel" below and login.',
  INVALID_DELETE_ACCOUNT: 'Please type "delete" to confirm you would like to delete your account',
  LIMIT_EXCEEDED:
    'You have reached your login attempt limit. Please allow some time and try again.',
  INTERNAL_ERROR:
    'Please try your request again. If unsuccessful on subsequent attempts, please contact customer service.',
};
