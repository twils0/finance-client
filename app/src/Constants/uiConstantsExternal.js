export const actionTypes = {
  SET_CURRENT_FORM: 'EXTERNAL_SET_CURRENT_FORM',
  SET_FORM_HEIGHT: 'EXTERNAL_SET_FORM_HEIGHT',
  SET_INPUT_VALUE_ERROR: 'EXTERNAL_SET_INPUT_VALUE_ERROR',
  SET_BUTTON_TEXT: 'EXTERNAL_SET_BUTTON_TEXT',
};

export const formNames = {
  RESET_PASSWORD: 'resetPasswordForm',
  FORGOT_PASSWORD: 'forgotPasswordForm',
  LOGIN: 'loginForm',
  CARD: 'cardForm',
  INFO: 'infoForm',
  SIGN_UP: 'signUpForm',
  CODE_MFA_PHONE: 'codePhoneMFAForm',
  CODE_VERIFY_PHONE: 'codePhoneVerifyForm',
  CODE_VERIFY_EMAIL: 'codeEmailVerifyForm',
  DEVICE: 'deviceForm',
};

export const inputNames = {
  [formNames.RESET_PASSWORD]: {
    CODE: 'codeResetPassword',
    PASSWORD: 'passwordResetPassword',
    PASSWORD2: 'password2ResetPassword',
  },
  [formNames.FORGOT_PASSWORD]: {
    EMAIL: 'loginPasswordEmail',
  },
  [formNames.LOGIN]: {
    EMAIL: 'loginEmail',
    PASSWORD: 'loginPassword',
  },
  [formNames.CARD]: {
    NAME_ON_CARD: 'cardNameOnCard',
    PROMO_CODE: 'cardPromoCode',
    STRIPE: 'cardStripe',
  },
  [formNames.INFO]: {
    NAME: 'infoName',
    PHONE: 'infoPhone',
    EMAIL: 'infoEmail',
  },
  [formNames.SIGN_UP]: {
    PASSWORD: 'signUpPassword',
    PASSWORD2: 'signUpPassword2',
  },
  [formNames.CODE_MFA_PHONE]: {
    CODE: 'codeMFAPhoneCode',
  },
  [formNames.CODE_VERIFY_PHONE]: {
    CODE: 'codeVerifyPhoneCode',
  },
  [formNames.CODE_VERIFY_EMAIL]: {
    CODE: 'codeVerifyEmailCode',
  },
};

export const buttonNames = {
  FIRST: 'firstButton',
  SECOND: 'secondButton',
  FORGOT_PASSWORD: 'forgotPasswordButton',
  RESEND_PHONE: 'resendPhoneButton',
  RESEND_EMAIL: 'resendEmailButton',
};

export const buttonTexts = {
  RESET: 'Reset',
  CANCEL: 'Cancel',
  SEND_CODE: 'Send Code',
  RESEND_PHONE: 'Resend SMS',
  RESEND_EMAIL: 'Resend Email',
  LOG_IN: 'Log In',
  FORGOT_PASSWORD: 'Forgot Password',
  SIGN_UP: 'Sign Up',
  NEXT: 'Next',
  PREVIOUS: 'Previous',
  SUBMIT: 'Submit',
  YES: 'Yes',
  NO: 'No',
  CONTINUE: 'Continue',
};
