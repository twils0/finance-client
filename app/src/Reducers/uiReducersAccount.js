import { combineReducers } from 'redux';

import {
  actionTypes,
  formNames,
  buttonNames,
  buttonTexts,
  inputNames,
} from '../Constants/uiConstantsAccount';

export const initialState = {
  utilities: {
    pageBody: {},
    list: {
      springSettings: { stiffness: 300, damping: 50 },
      width: 300,
      widthBuffer: 60,
      heightBuffer: 60,
      rowWidthIndent: 10,
      rowHeight: 50,
      rowHeightBufferBottom: 10,
    },
    separatorLine: {
      width: 4,
      height: 320,
      heightBuffer: 40,
    },
    forms: {
      springSettings: { stiffness: 200, damping: 26 },
      width: 400,
      widthBuffer: 60,
      rowWidthIndent: 10,
      headerWidthBufferRight: 10,
      formArrayFormsHeight: 350,
      formsHeightBuffer: 60,
      rowHeight: 60,
      rowHeightBuffer: 18,
    },
    buttons: {
      springSettings: { stiffness: 170, damping: 26 },
      width: 100,
      widthBuffer: 10,
      heightBuffer: 60,
    },
  },
  forms: {
    current: formNames.PROFILE,
    list: [...Object.values(formNames)],
    [formNames.CODE]: {
      id: formNames.CODE,
      height: 0,
      edit: false,
      inputsList: [...Object.values(inputNames[formNames.CODE])],
      inputs: {
        [inputNames[formNames.CODE].CODE_PHONE]: {
          parentId: formNames.CODE,
          id: inputNames[formNames.CODE].CODE_PHONE,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.CODE].CODE_EMAIL]: {
          parentId: formNames.CODE,
          id: inputNames[formNames.CODE].CODE_EMAIL,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.PROFILE]: {
      id: formNames.PROFILE,
      height: 240,
      edit: false,
      inputsList: [...Object.values(inputNames[formNames.PROFILE])],
      inputs: {
        [inputNames[formNames.PROFILE].NAME]: {
          parentId: formNames.PROFILE,
          id: inputNames[formNames.PROFILE].NAME,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.PROFILE].EMAIL]: {
          parentId: formNames.PROFILE,
          id: inputNames[formNames.PROFILE].EMAIL,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.PROFILE].EMAIL_ADDITIONAL]: {
          parentId: formNames.PROFILE,
          id: inputNames[formNames.PROFILE].EMAIL_ADDITIONAL,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.PROFILE].PHONE]: {
          parentId: formNames.PROFILE,
          id: inputNames[formNames.PROFILE].PHONE,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.BILLING]: {
      id: formNames.BILLING,
      height: 0,
      edit: false,
      inputsList: [...Object.values(inputNames[formNames.BILLING])],
      inputs: {
        [inputNames[formNames.BILLING].NAME_ON_CARD]: {
          parentId: formNames.BILLING,
          id: inputNames[formNames.BILLING].NAME_ON_CARD,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.BILLING].PROMO_CODE]: {
          parentId: formNames.BILLING,
          id: inputNames[formNames.BILLING].PROMO_CODE,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.BILLING].STRIPE]: {
          parentId: formNames.BILLING,
          id: inputNames[formNames.BILLING].STRIPE,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.CHANGE_PASSWORD]: {
      id: formNames.CHANGE_PASSWORD,
      height: 0,
      inputsList: [...Object.values(inputNames[formNames.CHANGE_PASSWORD])],
      inputs: {
        [inputNames[formNames.CHANGE_PASSWORD].OLD_PASSWORD]: {
          parentId: formNames.CHANGE_PASSWORD,
          id: inputNames[formNames.CHANGE_PASSWORD].OLD_PASSWORD,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD]: {
          parentId: formNames.CHANGE_PASSWORD,
          id: inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD2]: {
          parentId: formNames.CHANGE_PASSWORD,
          id: inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD2,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.DELETE_ACCOUNT]: {
      id: formNames.DELETE_ACCOUNT,
      height: 0,
      edit: false,
      inputsList: [...Object.values(inputNames[formNames.DELETE_ACCOUNT])],
      inputs: {
        [inputNames[formNames.DELETE_ACCOUNT].CONFIRM]: {
          parentId: formNames.DELETE_ACCOUNT,
          id: inputNames[formNames.DELETE_ACCOUNT].CONFIRM,
          value: '',
          errorMessage: '',
        },
      },
    },
  },
  buttons: {
    list: [...Object.values(buttonNames)],
    [buttonNames.FIRST]: { id: buttonNames.FIRST, text: buttonTexts.EDIT, visible: true },
    [buttonNames.SECOND]: { id: buttonNames.SECOND, text: buttonTexts.CANCEL, visible: false },
  },
};

initialState.utilities.pageBody.mediaWidth =
  50 +
  initialState.utilities.list.width +
  initialState.utilities.list.widthBuffer +
  initialState.utilities.separatorLine.width +
  initialState.utilities.forms.width +
  initialState.utilities.forms.widthBuffer;

export const utilities = (state = initialState.utilities) => state;

export const forms = (state = initialState.forms, action) => {
  switch (action.type) {
    case actionTypes.SET_CURRENT_FORM:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.SET_FORM_HEIGHT:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.SET_FORM_EDIT:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.SET_INPUT_VALUE_ERROR:
      return {
        ...state,
        [action.payload.parentId]: {
          ...state[action.payload.parentId],
          inputs: {
            ...state[action.payload.parentId].inputs,
            [action.payload.id]: {
              ...state[action.payload.parentId].inputs[action.payload.id],
              ...action.payload,
            },
          },
        },
      };
    default:
      return state;
  }
};

export const buttons = (state = initialState.buttons, action) => {
  switch (action.type) {
    case actionTypes.SET_BUTTON_TEXT:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.SET_BUTTON_VISIBLE:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    default:
      return state;
  }
};

const uiReducersAccount = combineReducers({
  utilities,
  forms,
  buttons,
});

export default uiReducersAccount;
