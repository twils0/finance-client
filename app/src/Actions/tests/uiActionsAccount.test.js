import configureMockStore from 'redux-mock-store';
import { initialState } from '../../Reducers/uiReducersAccount';
import { actionTypes } from '../../Constants/uiConstantsAccount';
import {
  setCurrentForm,
  setFormHeight,
  setFormEdit,
  setInputValueError,
  setButtonText,
  setButtonVisible,
} from '../uiActionsAccount';

const middleware = [];
const mockStore = configureMockStore(middleware);

describe('uiActionsAccount', () => {
  describe('setCurrentForm', () => {
    it("fails when not provided a payload with a 'current' key", () => {
      const store = mockStore(initialState);

      const payload = {};

      expect(() => store.dispatch(setCurrentForm(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { current: 'testId' };
      const expectedActions = [
        {
          type: actionTypes.SET_CURRENT_FORM,
          payload,
        },
      ];

      store.dispatch(setCurrentForm(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setFormHeight', () => {
    it("fails when not provided a payload with a 'id' key", () => {
      const store = mockStore(initialState);

      const payload = { height: 1 };

      expect(() => store.dispatch(setFormHeight(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'height' key", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId' };

      expect(() => store.dispatch(setFormHeight(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with an 'height' key that is not a number", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', height: 'height' };

      expect(() => store.dispatch(setFormHeight(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', height: 1 };
      const expectedActions = [
        {
          type: actionTypes.SET_FORM_HEIGHT,
          payload,
        },
      ];

      store.dispatch(setFormHeight(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setFormEdit', () => {
    it("fails when not provided a payload with a 'id' key", () => {
      const store = mockStore(initialState);

      const payload = { edit: false };

      expect(() => store.dispatch(setFormEdit(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'edit' key", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId' };

      expect(() => store.dispatch(setFormEdit(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with an 'edit' key that is not a boolean", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', edit: 'edit' };

      expect(() => store.dispatch(setFormEdit(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', edit: false };
      const expectedActions = [
        {
          type: actionTypes.SET_FORM_EDIT,
          payload,
        },
      ];

      store.dispatch(setFormEdit(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setInputValueError', () => {
    it("fails when not provided a payload with a 'id' key", () => {
      const store = mockStore(initialState);

      const payload = {
        parentId: 'testParentId',
        value: 'testValue',
        errorMessage: 'testErrorMessage',
      };

      expect(() => store.dispatch(setInputValueError(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'parentId' key", () => {
      const store = mockStore(initialState);

      const payload = {
        id: 'testId',
        value: 'testValue',
        errorMessage: 'testErrorMessage',
      };

      expect(() => store.dispatch(setInputValueError(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'value' key", () => {
      const store = mockStore(initialState);

      const payload = {
        id: 'testId',
        parentId: 'testParentId',
        errorMessage: 'testErrorMessage',
      };

      expect(() => store.dispatch(setInputValueError(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'errorMessage' key", () => {
      const store = mockStore(initialState);

      const payload = {
        id: 'testId',
        parentId: 'testParentId',
        value: 'testValue',
      };

      expect(() => store.dispatch(setInputValueError(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = {
        id: 'testId',
        parentId: 'testParentId',
        value: 'testValue',
        errorMessage: 'testErrorMessage',
      };
      const expectedActions = [
        {
          type: actionTypes.SET_INPUT_VALUE_ERROR,
          payload,
        },
      ];

      store.dispatch(setInputValueError(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setButtonText', () => {
    it("fails when not provided a payload with a 'id' key", () => {
      const store = mockStore(initialState);

      const payload = { text: 'testText' };

      expect(() => store.dispatch(setButtonText(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'text' key", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId' };

      expect(() => store.dispatch(setButtonText(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', text: 'testText' };
      const expectedActions = [
        {
          type: actionTypes.SET_BUTTON_TEXT,
          payload,
        },
      ];

      store.dispatch(setButtonText(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setButtonVisible', () => {
    it("fails when not provided a payload with a 'id' key", () => {
      const store = mockStore(initialState);

      const payload = { visible: false };

      expect(() => store.dispatch(setButtonVisible(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'visible' key", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId' };

      expect(() => store.dispatch(setButtonVisible(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with an 'visible' key that is not a boolean", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', visible: 'visible' };

      expect(() => store.dispatch(setButtonVisible(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', visible: false };
      const expectedActions = [
        {
          type: actionTypes.SET_BUTTON_VISIBLE,
          payload,
        },
      ];

      store.dispatch(setButtonVisible(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });
});
