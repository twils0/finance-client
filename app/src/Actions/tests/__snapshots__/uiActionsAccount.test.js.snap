// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiActionsAccount setButtonText fails when not provided a payload with a 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"text\\":\\"testText\\"}"`;

exports[`uiActionsAccount setButtonText fails when not provided a payload with a 'text' key 1`] = `"Please enter a value for the 'text' key - {\\"id\\":\\"testId\\"}"`;

exports[`uiActionsAccount setButtonVisible fails when not provided a payload with a 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"visible\\":false}"`;

exports[`uiActionsAccount setButtonVisible fails when not provided a payload with a 'visible' key 1`] = `"Please enter a value for the 'visible' key - {\\"id\\":\\"testId\\"}"`;

exports[`uiActionsAccount setButtonVisible fails when provided a payload with an 'visible' key that is not a boolean 1`] = `"Please enter a boolean value for the 'visible' key - {\\"id\\":\\"testId\\",\\"visible\\":\\"visible\\"}"`;

exports[`uiActionsAccount setCurrentForm fails when not provided a payload with a 'current' key 1`] = `"Please enter a value for the 'current' key - {}"`;

exports[`uiActionsAccount setFormEdit fails when not provided a payload with a 'edit' key 1`] = `"Please enter a value for the 'edit' key - {\\"id\\":\\"testId\\"}"`;

exports[`uiActionsAccount setFormEdit fails when not provided a payload with a 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"edit\\":false}"`;

exports[`uiActionsAccount setFormEdit fails when provided a payload with an 'edit' key that is not a boolean 1`] = `"Please enter a boolean value for the 'edit' key - {\\"id\\":\\"testId\\",\\"edit\\":\\"edit\\"}"`;

exports[`uiActionsAccount setFormHeight fails when not provided a payload with a 'height' key 1`] = `"Please enter a value for the 'height' key - {\\"id\\":\\"testId\\"}"`;

exports[`uiActionsAccount setFormHeight fails when not provided a payload with a 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"height\\":1}"`;

exports[`uiActionsAccount setFormHeight fails when provided a payload with an 'height' key that is not a number 1`] = `"Please enter a number value for the 'height' key - {\\"id\\":\\"testId\\",\\"height\\":\\"height\\"}"`;

exports[`uiActionsAccount setInputValueError fails when not provided a payload with a 'errorMessage' key 1`] = `"Please enter a value for the 'errorMessage' key - {\\"id\\":\\"testId\\",\\"parentId\\":\\"testParentId\\",\\"value\\":\\"testValue\\"}"`;

exports[`uiActionsAccount setInputValueError fails when not provided a payload with a 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"parentId\\":\\"testParentId\\",\\"value\\":\\"testValue\\",\\"errorMessage\\":\\"testErrorMessage\\"}"`;

exports[`uiActionsAccount setInputValueError fails when not provided a payload with a 'parentId' key 1`] = `"Please enter a value for the 'parentId' key - {\\"id\\":\\"testId\\",\\"value\\":\\"testValue\\",\\"errorMessage\\":\\"testErrorMessage\\"}"`;

exports[`uiActionsAccount setInputValueError fails when not provided a payload with a 'value' key 1`] = `"Please enter a value for the 'value' key - {\\"id\\":\\"testId\\",\\"parentId\\":\\"testParentId\\",\\"errorMessage\\":\\"testErrorMessage\\"}"`;
